### PR TITLE
Handle Hiredis silly RuntimeError on DNS resolution issues

### PR DESCRIPTION
**Pull request by byroot**
WIP, need to add tests.

cc @fw42

### CODE REVIEW

**Comment by fw42** [comment]
Added a test and removed the special case for hiredis. This is stupid. We should fix that one upstream (in hiredis-rb). Will look into it. For now, this PR solves the problem for the ruby redis driver, which I think is an improvement.

@byroot, cool?

**Comment by Sirupsen** [comment]
LGTM

File: test/redis_test.rb [hunk 1]
```diff
@@ -51,6 +51,29 @@ def test_connection_errors_open_the_circuit
     end
   end
 
+  def test_dns_resolution_errors_in_ruby_driver_open_the_circuit
+    redis_options = { host: "invalid-dns-name", semian: SEMIAN_OPTIONS, reconnect_attempts: 0 }
+
+    clients = [
+      Redis.new(driver: 'ruby', **redis_options),
+      # Redis.new(driver: 'hiredis', **redis_options)
```
Comment (byroot) [inline test/redis_test.rb]: So you don't want to test hiredis? Even though I agree it should be fixed upstream, we have no idea what the ETA might be.

**Comment by byroot** [comment]
Not sure about the ignoring hiredis situation. But I'll let you decide.

Regular Redis LGTM.

:shipit:

File: test/redis_test.rb (hunk 1 above)
Comment (fw42) [inline test/redis_test.rb]: well the test is gonna fail :) and I don't like that ugly hack where we look at the message :/

**Comment by byroot** [comment]
Oh, you have a rubocop violation :/ Sorry.

**Comment by fw42** [comment]
Do you remember why this never shipped?

**Comment by fw42** [comment]
Scott is trying to do something similar in https://github.com/Shopify/semian/pull/184. Which one should we ship?

**Comment by casperisfine** [comment]
Sorry can't remember why this one fall through the cracks.

Scott's look cleaner anyway. Closing.

**Comment by fw42** [comment]
The test here (assert the circuit was opened) seems useful, maybe we should port that to the other PR?

**Pull request closed without merging**